ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/TraceBlock.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.BlockParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.BlockTracer;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.TransactionTrace;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.tracing.flat.FlatTraceGenerator;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.tracing.flat.RewardTraceGenerator;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.debug.TraceOptions;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.vm.DebugOperationTracer;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+public class TraceBlock extends AbstractBlockParameterMethod {
+  private final Supplier<BlockTracer> blockTracerSupplier;
+  private final ProtocolSchedule<?> protocolSchedule;
+
+  public TraceBlock(
+      final Supplier<BlockTracer> blockTracerSupplier,
+      final ProtocolSchedule<?> protocolSchedule,
+      final BlockchainQueries queries) {
+    super(queries);
+    this.blockTracerSupplier = blockTracerSupplier;
+    this.protocolSchedule = protocolSchedule;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.TRACE_BLOCK.getMethodName();
+  }
+
+  @Override
+  protected BlockParameter blockParameter(final JsonRpcRequestContext request) {
+    return request.getRequiredParameter(0, BlockParameter.class);
+  }
+
+  @Override
+  protected Object resultByBlockNumber(
+      final JsonRpcRequestContext request, final long blockNumber) {
+    if (blockNumber == BlockHeader.GENESIS_BLOCK_NUMBER) {
+      // Nothing to trace for the genesis block
+      return emptyResult();
+    }
+
+    return getBlockchainQueries()
+        .getBlockchain()
+        .getBlockByNumber(blockNumber)
+        .map(this::traceBlock)
+        .orElse(null);
+  }
+
+  private Object traceBlock(final Block block) {
+
+    if (block == null) {
+      return emptyResult();
+    }
+    final ObjectMapper mapper = new ObjectMapper();
```

Comment:
you can make the mapper static - also reuse the same one with `emptyResult`

Reply:
Done.